core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -588,19 +595,21 @@ private Evaluator extractInclusiveDeleteExpression(ManifestReader reader) {
             outputManifests.add(bin.get(0));
             return;
           }
-
           // if the bin has a new manifest (the new data files) then only merge it if the number of
           // manifests is above the minimum count. this is applied only to bins with an in-memory
           // manifest so that large manifests don't prevent merging older groups.
           if (bin.contains(cachedNewManifest) && bin.size() < minManifestsCountToMerge) {
             // not enough to merge, add all manifest files to the output list
             outputManifests.addAll(bin);
+          } else if (bin.contains(firstAppendedManifest) && bin.size() < minManifestsCountToMerge) {
```

Comment:
This logic could lead to two unmerged bins if there are new files and appended manifests. If the new files create a manifest in its own bin, then the next bin would contain the first appended manifest. I think you should merge the two conditions:

```java
if ((bin.contains(cachedNewManifest) || bin.contains(firstAppendedManifest)) && bin.size < minManifestCountToMerge) { ... }
```

Reply:
true, update with OR condition.